File: 2-add-two-numbers/add-two-numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* ptr1=l1;
        ListNode* ptr2=l2;
        ListNode* head=new ListNode(-1);
        ListNode* ptr=head;
        int carry=0;
        while(ptr1!=NULL && ptr2!=NULL){
            int val=carry+ptr1->val+ptr2->val;
            carry=val/10;
            val=val%10;
            ListNode* newNode=new ListNode(val);
            ptr->next=newNode;
            ptr=ptr->next;
            ptr1=ptr1->next;
            ptr2=ptr2->next;
        }
        while(ptr1!=NULL){
            int val=carry+ptr1->val;
            carry=val/10;
            val=val%10;
            ListNode* newNode=new ListNode(val);
            ptr->next=newNode;
            ptr=ptr->next;
            ptr1=ptr1->next;
        }
        while(ptr2!=NULL){
            int val=carry+ptr2->val;
            carry=val/10;
            val=val%10;
            ListNode* newNode=new ListNode(val);
            ptr->next=newNode;
            ptr=ptr->next;
            ptr2=ptr2->next;
        }
        if(carry!=0){
            ListNode* newNode=new ListNode(carry);
            ptr->next=newNode;
            ptr=ptr->next;
        }
        return head->next;
    }
};

File: 2-add-two-numbers/add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        ptr1=l1; ptr2=l2; head=ListNode(-1); ptr=head; carry=0
        while ptr1!=None and ptr2!=None:
            val=carry+ptr1.val+ptr2.val
            carry=val//10
            val%=10
            newNode=ListNode(val)
            ptr.next=newNode
            ptr=ptr.next
            ptr1=ptr1.next
            ptr2=ptr2.next
        while ptr1!=None:
            val=carry+ptr1.val
            carry=val//10
            val%=10
            newNode=ListNode(val)
            ptr.next=newNode
            ptr=ptr.next
            ptr1=ptr1.next
        while ptr2!=None:
            val=carry+ptr2.val
            carry=val//10
            val%=10
            newNode=ListNode(val)
            ptr.next=newNode
            ptr=ptr.next
            ptr2=ptr2.next
        if carry!=0:
            newNode=ListNode(carry)
            ptr.next=newNode
            
        return head.next
        